examples/00-basic/13-get_data_by_id.py

```diff
@@ -0,0 +1,173 @@
+print(disp_selection_fc)
+
+###############################################################################
+# Note how the number of entities has decreased from 15113 to 12970 entities
+# after applying the scoping on ``my_named_selection``.
+
+###############################################################################
+# Get the only Field available
+disp_selection = disp_selection_fc[0]
+
+###############################################################################
+# Scope mesh only for that named selection
+mesh_from_scoping_op = ops.mesh.from_scoping()
+mesh_from_scoping_op.inputs.scoping.connect(mesh_scoping)
+mesh_from_scoping_op.inputs.mesh.connect(model.metadata.meshed_region)
+mesh_selection = mesh_from_scoping_op.outputs.mesh()
+
+###############################################################################
+# Plot ``mesh_selection``
+mesh_selection.plot()
+
+###############################################################################
+# The plot shows only one of the two cubes from the total geometry shown before.
+
+###############################################################################
+# Identify the location of the Field
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Check the location of the scoping to see what is the information associated
+# to the IDs. The displacement field is defined at ``"nodal"`` location. Other
+# fields may have other locations such as ``"elemental"`` or ``"elementalNodal"``.
+# Note that this way of retrieving data by ID is only valid for locations that
+# are either ``"nodal"`` or ``"elemental"``.
+print(disp_selection.scoping.location)
+
+###############################################################################
+# Another way to check that ``disp_selection`` contains nodal data is to check
+# that the number of entities in the displacement FieldsContainer (12970)
+# matches the number of nodes of the scoped mesh ``mesh_selection``
+print(mesh_selection.nodes.n_nodes)
+
+###############################################################################
+# Instead, the number of elements does not match
+print(mesh_selection.elements.n_elements)
+
+###############################################################################
+# Access data in ``disp_selection`` by ID (index approach)
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Get the list of nodal IDs
+ids = disp_selection.scoping.ids
+
+###############################################################################
+# Retrieve data of ``disp_selection`` from the server. Note that this is done
+# before the loop below so only one call is made to the server.
+start = time.time()
+data = disp_selection.data
+
+###############################################################################
+# Loop over ``ids`` to have access to both, the node ID and the displacement
+# for that node
+nodal_disp_index = np.zeros([mesh_selection.nodes.n_nodes, 3])
+for idx, node_id in enumerate(ids):
+    node_disp = data[idx]
```

Comment:
@Guillem, this is true when you have nodal or elemental results. When you read stresses for example as Elemental nodal data, then you will need to use field.get_entity_data(idx) instead of data[idx], otherwise you will get only the first stress tensor, this is also true for layered results. 
We may need to add a similar loop with field.get_entity_data(); when the server is in in-process mode I think that you will not have overhead, if the server is remote then you may need to use as_local_field option and field.get_entity_data()